frontend/catalyst/jax_primitives.py

```diff
@@ -306,6 +311,55 @@ class MeasurementPlane(Enum):
 quantum_kernel_p.multiple_results = True
 measure_in_basis_p = Primitive("measure_in_basis")
 measure_in_basis_p.multiple_results = True
+quantum_subroutine_p = copy.deepcopy(pjit_p)
+quantum_subroutine_p.name = "quantum_subroutine_p"
+
+subroutine_cache: dict[callable, callable] = {}
+
+
+def subroutine(func):
+    """
+    This is technical debt. pjit_p is the only primitive that will not hoist
+    jaxpr_consts outside of its definition. Here, we are creating a new primitive
+    called quantum_subroutine_p that is essentially equivalent to pjit_p.
+
+    The problem with this approach is that it will also change the definitions of jax.jit
+    which may be purely classical. The best solutions would be to have two primitives.
+
+    But in order to have two primitives, we would need to replace pjit_p selectively,
+    depending on whether or not we are inside of a qml.qjit context. This would involve
+    rewriting some methods inside JAX or copying them and modifying that one key aspect.
+
+    I, @erick-xanadu, am currently unsure about what is the best way to do this.
+    Maybe this is sufficiently good?
```

Comment:
I don't know why but this made me laugh 😆 It sounds like you're about to make a solemn declaration